### PR TITLE
Spring / Jersey upgrade

### PR DESCRIPTION
**Pull request by dankraw**

### CODE REVIEW

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/auth/AuthorizationFilter.java [hunk 1]
```diff
@@ -3,15 +3,13 @@
...
-import javax.ws.rs.Priorities;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
 import javax.ws.rs.ext.Provider;
 import java.io.IOException;
 
-
 @Provider
-@Priority(Priorities.AUTHORIZATION)
+@Priority(1999) // fixing equal values reordering issue of Jersey's 2.23.2 RankedComparator (Priorities.AUTHORIZATION=2000)
```
Comment (dankraw) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/auth/AuthorizationFilter.java]: I've made a PR for this: https://github.com/jersey/jersey/pull/239

Comment (wojtkiewicz) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/auth/AuthorizationFilter.java]: good job

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/env/SpringlessServiceIteratorProvider.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.glassfish.jersey.internal.ServiceFinder;
+
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+
+class SpringlessServiceIteratorProvider extends ServiceFinder.ServiceIteratorProvider {
+
+    private static final ServiceFinder.ServiceIteratorProvider delegate = new ServiceFinder.DefaultServiceIteratorProvider();
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/env/SpringlessServiceIteratorProvider.java]: \[Checkstyle\] ERROR: Name 'delegate' must match pattern '^\[A\-Z\]\[A\-Z0\-9\]\*\(\_\[A\-Z0\-9\]\+\)\*|logger\*$'\.

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/env/SpringlessServiceIteratorProvider.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @Override
+    public <T> Iterator<T> createIterator(Class<T> service, String serviceName, ClassLoader loader, boolean ignoreOnClassNotFound) {
+        Iterator<T> iterator = delegate.createIterator(service, serviceName, loader, ignoreOnClassNotFound);
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED), false)
+                .filter(o -> !o.getClass().getCanonicalName().equals("org.glassfish.jersey.server.spring.SpringComponentProvider"))
+                .iterator();
+    }
+
+    @Override
+    public <T> Iterator<Class<T>> createClassIterator(Class<T> service, String serviceName, ClassLoader loader, boolean ignoreOnClassNotFound) {
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/env/SpringlessServiceIteratorProvider.java]: \[Checkstyle\] INFO: Line is longer than 140 characters \(found 144\)\.

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/helper/GraphiteEndpoint.java [hunk 4]
```diff
@@ -22,9 +22,9 @@
             + "{\"target\": \"sumSeries(stats.tech.hermes.consumer.*.meter.SUBSCRIPTION.m1_rate)\", \"datapoints\": [[RATE, TIMESTAMP]]}"
             + "]";
 
-    private static final String TOPIC_URL_PATTERN = "/.*sumSeries\\(stats.tech.hermes\\.(consumer|producer)\\.\\*\\.meter\\.[^\\.]*\\.[^\\.]*\\.m1_rate\\).*";
+    private static final String TOPIC_URL_PATTERN = "/.*sumSeries%28stats.tech.hermes\\.(consumer|producer)\\.%2A\\.meter\\.[^\\.]*\\.[^\\.]*\\.m1_rate%29.*";
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/helper/GraphiteEndpoint.java]: \[Checkstyle\] INFO: Line is longer than 140 characters \(found 158\)\.

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/helper/GraphiteEndpoint.java [hunk 5]
```diff
@@ -22,9 +22,9 @@
...
 
-    private static final String SUBSCRIPTION_URL_PATTERN = "/.*sumSeries\\(stats.tech.hermes\\.consumer\\.\\*\\.meter\\.[^\\.]*\\.[^\\.]*\\.[^\\.]*\\.m1_rate\\).*";
+    private static final String SUBSCRIPTION_URL_PATTERN = "/.*sumSeries%28stats.tech.hermes\\.consumer\\.%2A\\.meter\\.[^\\.]*\\.[^\\.]*\\.[^\\.]*\\.m1_rate%29.*";
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/helper/GraphiteEndpoint.java]: \[Checkstyle\] INFO: Line is longer than 140 characters \(found 164\)\.

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/env/SpringlessServiceIteratorProvider.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.glassfish.jersey.internal.ServiceFinder;
+import org.glassfish.jersey.server.spring.SpringComponentProvider;
+
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+
+class SpringlessServiceIteratorProvider extends ServiceFinder.ServiceIteratorProvider {
```
Comment (sputnikci) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/env/SpringlessServiceIteratorProvider.java]: \[Checkstyle\] ERROR: Class SpringlessServiceIteratorProvider should be declared as final\.

**Pull request merged**